src/xo-app/jobs/helpers/index.js

```diff
@@ -0,0 +1,115 @@
+import find from 'lodash/find'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import Select from 'react-select'
+import { propTypes } from 'utils'
+
+@propTypes({
+  autoFocus: propTypes.bool,
+  defaultValue: propTypes.any,
+  disabled: propTypes.bool,
+  multi: propTypes.bool,
+  onChange: propTypes.func,
+  options: propTypes.array,
+  placeholder: propTypes.string,
+  predicate: propTypes.func,
+  required: propTypes.bool
+})
+export class SelectPlainObject extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      value: this._setValue(props.defaultValue, props)
```

Comment:
I'm not fond of these `_set*()` functions which do not set anything and simply return a value…

I would like to change this pattern, either by setting the state directly (working around the different behaviors between the constructor and other lifecycle methods) or renaming them (e.g. `_computeValue()`).